doc/source/data/key-concepts.rst

```diff
@@ -36,75 +37,72 @@ This flexibility is a unique characteristic of Ray Datasets.
+Datasets use either Ray tasks or Ray actors to transform datasets. By default, Datasets uses tasks.
+
+To use Actors, pass an :class:`ActorPoolStrategy` to ``compute`` in methods like
+:meth:`~ray.data.Dataset.map_batches`. :class:`ActorPoolStrategy` creates an autoscaling
+pool of Ray actors. This allows you to cache expensive state initialization
+(e.g., for GPU-based tasks).
 
 .. image:: images/dataset-map.svg
    :align: center
 ..
   https://docs.google.com/drawings/d/12STHGV0meGWfdWyBlJMUgw7a-JcFPu9BwSOn5BjRw9k/edit
 
-See the :ref:`Transforming Datasets Guide <transforming_datasets>` for an in-depth guide
-on transforming datasets.
+For an in-depth guide on transforming datasets, read :ref:`Transforming Datasets <transforming_datasets>`.
 
 Shuffling Data
 ==============
 
-Certain operations like *sort* or *groupby* require data blocks to be partitioned by value, or *shuffled*. Datasets uses tasks to implement distributed shuffles in a map-reduce style, using map tasks to partition blocks by value, and then reduce tasks to merge co-partitioned blocks together.
+Operations like :meth:`~ray.data.Dataset.sort` and :meth:`~ray.data.Dataset.groupby`
+require blocks to be partitioned by value or *shuffled*. Datasets uses tasks to shuffle blocks in a map-reduce
+style: map tasks partition blocks by value and then reduce tasks merge co-partitioned
+blocks.
 
-You can also change just the number of blocks of a Dataset using :meth:`~ray.data.Dataset.repartition`.
+Call :meth:`~ray.data.Dataset.repartition` to change the number of blocks.
 Repartition has two modes:
 
-1. ``shuffle=False`` - performs the minimal data movement needed to equalize block sizes
-2. ``shuffle=True`` - performs a full distributed shuffle
+* ``shuffle=False`` - performs the minimal data movement needed to equalize block sizes
+* ``shuffle=True`` - performs a full distributed shuffle
 
 .. image:: images/dataset-shuffle.svg
    :align: center
 
 ..
   https://docs.google.com/drawings/d/132jhE3KXZsf29ho1yUdPrCHB9uheHBWHJhDQMXqIVPA/edit
 
-Datasets shuffle can scale to processing hundreds of terabytes of data. See the :ref:`Performance Tips Guide <shuffle_performance_tips>` for an in-depth guide on shuffle performance.
+Datasets can shuffle hundreds of terabytes of data. For an in-depth guide on shuffle performance, read :ref:`Performance Tips and Tuning <shuffle_performance_tips>`.
 
 Execution mode
 ==============
 
-Most Datasets operations are lazily executed. Operations won't be executed until the dataset is consumed or :meth:`ds.fully_executed() <ray.data.Dataset.fully_executed>` is called to manually trigger execution. See the :ref:`Execution Section <datasets_execution>` for more details.
+Datasets is lazy. The library doesn't execute operations until you
```

Comment:
plural?

```suggestion
Datasets are lazy. The library doesn't execute operations until you
```

Reply:
cc @jianoaix to confirm if all operations are now lazy...I think there are still operations that are not.

we should explicitly list the ones that are not.

Reply:
Meant Datasets as in the library, but Datasets as in the Python object might make more sense.

Reply:
We are using "operations" to cover both transformation and consumption, so only the transformations are meant to be lazy (consumption calls are to trigger the execution), and they are lazy by default already. There are exceptions, like `Union` is still eager.

Reply:
Got it, can we explicitly list out the current exceptions?

Maybe something similar to the @ConsumptionAPI annotations that we have for the current API?

Reply:
Could we list exceptions and add a decorator in a follow-up PR?

Reply:
Keeping it as "most transformations are lazy" is fine